### PR TITLE
Add build files to build cloud image and manifests to run in k8s cluster

### PR DESCRIPTION
**Pull request by nkwangleiGIT**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Allow users to easily run edgecontroller inside k8s cluster

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
Will add the instruction about how to use kubectl to run edgecontroller in k8s cluster, and will also add doc about how to run edged in container later.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,3 +25,6 @@ edge_cross_build:
 edgecontroller:
 	cd cloud/edgecontroller && $(MAKE)
 
+.PHONY: cloudimage
+cloudimage:
+	docker build -t kubeedge/edgecontroller:v0.2 -f build/cloud/Dockerfile .
```
Comment (m1093782566) [inline Makefile]: make image tag configurable?

File: build/cloud/03-clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  resources:
+  - nodes
+  - configmaps
+  - pods
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  - update
```
Comment (m1093782566) [inline build/cloud/03-clusterrole.yaml]: Please add a newline please.

File: build/cloud/06-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
+apiVersion: v1
```
Comment (m1093782566) [inline build/cloud/06-secret.yaml]: Can we generate the cert files at runtime?

File: build/cloud/08-service.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    k8s-app: kubeedge
+    kubeedge: edgecontroller
+spec:
+  ports:
+  - port: 10000
+    name: cloudhub
+  selector:
+    k8s-app: kubeedge
+    kubeedge: edgecontroller
```
Comment (m1093782566) [inline build/cloud/08-service.yaml]: Add a newline please.

File: build/cloud/07-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    matchLabels:
+      k8s-app: kubeedge
+      kubeedge: edgecontroller
+  template:
+    metadata:
+      labels:
+        k8s-app: kubeedge
+        kubeedge: edgecontroller
+      annotations:
+        sidecar.istio.io/inject: "false"
```
Comment (rohitsardesai83) [inline build/cloud/07-deployment.yaml]: why is this annotation specified ?

Comment (nkwangleiGIT) [reply build/cloud/07-deployment.yaml]: We can remove this, it's just used to disable istio sidecar injection

Comment (rohitsardesai83) [reply build/cloud/07-deployment.yaml]: better we remove. thanks.

**Comment by tedli** [comment]
https://github.com/tenxcloud/kubeedge/pull/2

here is the up coming fixs.

**Comment by nkwangleiGIT** [comment]
Some questions here:
1）Is there any registry server that we can use to upload all KubeEdge images?
2）Is it enough to only mount docker.sock to run edged inside a container? for k8s kubelet, we have more to configure to make it run inside a container.

File: edge/pkg/eventbus/event_bus.go [hunk 6]
```diff
@@ -35,13 +36,24 @@ type eventbus struct {
 
 func init() {
 	mode := config.CONFIG.GetConfigurationByKey("mqtt.mode")
+	var mqttMode int
+	var ok bool
 	if mode == nil {
-		mode = internalMqttMode
+		mqttMode = internalMqttMode
+	} else {
+		if mqttMode, ok = mode.(int); !ok {
```
Comment (m1093782566) [inline edge/pkg/eventbus/event_bus.go]: @nkwangleiGIT 

Are you checking if mqttMode is an integer? Can we simply do something like:

```
i, err := strconv.Atoi("mqttMode")
if err != nil {
  panic("mqtt.mode should be an integer")
}
```

Comment (tedli) [reply edge/pkg/eventbus/event_bus.go]: The story is that, the original logic get the mqtt.mode from yaml, which the whole yaml content is parsed into
```golang
map[string]interface{}
```
so the underneath mqtt.mode value type is `int` what can cast to int value directly.

However, when deploying using container, we use environment variable to overwrite these config entries, and mqtt.mode parsed into string, so it panics when casting to int.

So adds the conditions to parse mqtt.mode no matter it is a string or int value.

**Comment by m1093782566** [comment]
@nkwangleiGIT 

> 1）Is there any registry server that we can use to upload all KubeEdge images?

Yes, we have created the kubeedge org and edgecontroller repository in dockerhub.

File: edge/pkg/eventbus/event_bus.go (hunk 6 above)
Comment (m1093782566) [reply edge/pkg/eventbus/event_bus.go]: Okay, we need some UTs for this change.

File: build/edge/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,27 @@
+FROM golang:1.12-alpine3.9 AS builder
```
Comment (m1093782566) [inline build/edge/Dockerfile]: @sids-b 

Did we test if edge-core can run inside a container?

**Comment by nkwangleiGIT** [comment]
> @nkwangleiGIT
> 
> > 1）Is there any registry server that we can use to upload all KubeEdge images?
> 
> Yes, we have created the kubeedge org and edgecontroller repository in dockerhub.

Can we get the write access to the repositories? or you have the owner to do this? We can start to simplify the document once the images are ready on dockerhub.

**Comment by m1093782566** [comment]
@nkwangleiGIT 

No worries, I will manage the docker registry.

**Comment by tedli** [comment]
> 2）Is it enough to only mount docker.sock to run edged inside a container? for k8s kubelet, we have more to configure to make it run inside a container.

Please mark this PR as `WIP` since edgecore can not fully functioned with just docker.sock mounted.

I'm working on it now.

**Comment by m1093782566** [comment]
@nkwangleiGIT @tedli 

I would be appreciated if you can cut this PR into smaller ones for the sake of easily review and merge.

**Comment by nkwangleiGIT** [comment]
@tedli OK, let's close this one and split it to some small PRs

**Pull request closed without merging**